src/torrent.cpp

```diff
@@ -159,6 +159,13 @@ namespace libtorrent {
 		, m_state(torrent_status::checking_resume_data)
 	{}
 
+	void torrent_hot_members::remove_connection(peer_connection const* p)
+	{
+		auto const i = sorted_find(m_connections, p);
```

Comment:
this function should assert ``TORRENT_ASSERT(m_iterating_connections == 0);``. In fact, it probably should not be a member of ``torrent_hot_members``, but just ``torrent``, right?